### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by ocrearyy**
In this project, I did the following: 

- Read the [Bookstore API documentation](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to learn how to use the API.
- Add redux-thunk to your project.
- Use [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) for making API calls.
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch list of books.
- Dispatch another action (add an action type) that will pass API response to reducer and updates the state.
- Refactor your add book and remove book features to persist your changes in the server.
- Change your existing Action Creators, so they return functions.
- Make changes by using API.
- Dispatch an action (action type that existed before refactor) that will updates the state accordingly.
- Add book and remove book should work in the same way after the refactor.
- No styling is required.

### CODE REVIEW

File: src/API/Api.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// const apiId = 'HCjJZpfpfCQz77Ezrye5';
```
Comment (ahmetbozaci) [inline src/API/Api.js]: - [x] For best practice do not commit old pieces of code as inline comments. [Requirement](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=Do%20not%20commit%20old%20pieces%20of%20code%20as%20inline%20comments.%20They%20will%20make%20your%20project%20look%20messy.%20If%20you%20need%20to%20review%20a%20previous%20version%20of%20your%20code%2C%20you%20can%20always%20use%20git%20history.)

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
 
-export const lessBooks = (id) => ({
-  type: REMOVE_BOOK, payload: id,
-});
+export const lessBooks = (id) => (dispatch) => {
+  removeBook(id).then((result) => {
+    console.log(result);
+    if (result.ok) dispatch({ type: REMOVE_BOOK, payload: id });
+  });
+};
```
Comment (ahmetbozaci) [inline src/redux/books/books.js]: - [ ] The remove book function is not working well. When clicking the removed book it is deleting all books please fix it.

https://user-images.githubusercontent.com/38635158/167261780-176c2c75-969d-4c14-91f5-3ea74eb7d80b.mp4

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @oshanedesign  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.




### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] To make a professional readme please fix the broken link in the license.

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/API/Api.js (hunk 1 above)
Comment (ocrearyy) [reply src/API/Api.js]: @ahmetbozaci  thank you, old piece of code removed 
e61

File: src/redux/books/books.js (hunk 2 above)
Comment (ocrearyy) [reply src/redux/books/books.js]: @ahmetbozaci thanks for pointing that out , issue fixed.

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**